packages/union-component/src/components/DonationPaymentForm.tsx

```diff
@@ -155,6 +176,11 @@ const DonationPaymentForm: React.FC<Props> = ({
             {errorMessage || DEFAULT_ERROR}
           </DonationWizard.ErrorText>
         ) : null}
+        {phoneNumberError ? (
```

Comment:
I would have expected to have the phone number error below the phone number input.
_Why are we adding the error here?_

Besides, I see some benefits into have state management within the phone input component. Because then this form component will only "known that the phone input will return a valid value"